install-openfaas.sh

```diff
@@ -13,33 +18,44 @@ cd faasd
 mkdir -p /var/lib/faasd/.docker/
 EOF
 
-chmod +x install-openfaas.sh
-./install-openfaas.sh >> install-openfaas.log 2>&1
-
-#installing docker
-cat <<EOF > install-docker.sh
-#!/bin/bash
-set -ex
-apt-get update
-apt-get install -y apt-transport-https ca-certificates curl software-properties-common
-curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
-echo "deb [signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
-apt-get update
-apt-get install -y docker-ce docker-ce-cli containerd.io
-usermod -aG docker root
-systemctl enable docker
-systemctl start docker
-EOF
-
-chmod +x install-docker.sh
-./install-docker.sh >> install-docker.log 2>&1
-
-#installing aws-cli
-cat <<EOF > install-aws-cli.sh
-#!/bin/bash
-snap install aws-cli --classic
-EOF
-
-chmod +x install-aws-cli.sh
-# ./install-aws-cli.sh >> install-aws-cli.log 2>&1
-
+    chmod +x install-openfaas.sh
+    ./install-openfaas.sh >>install-openfaas.log 2>&1
+}
+
+function docker_setup() {
+    apt-get -y update
+    apt-get -y install net-tools ca-certificates curl gnupg
+
+    if [ ! -f /etc/apt/keyrings/docker.gpg ]; then
+        install -m 0755 -d /etc/apt/keyrings
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+        chmod a+r /etc/apt/keyrings/docker.gpg
+        echo \
+            "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+        "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" |
+            tee /etc/apt/sources.list.d/docker.list >/dev/null
+    fi
+
+    apt-get -y update
+    apt-get -y install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+    dockerd >/dev/null 2>&1 </dev/null &
+    docker run hello-world
+    docker rm $(docker ps -a | grep hello | cut -d" " -f1)
+    docker rmi hello-world
+}
+
+function firewall_setup() {
+    if ! command -v ufw &>/dev/null; then
+        apt-get update
+        apt-get install ufw
+    fi
+
+    ufw -y enable
+    ufw allow 8080/tcp
+    ufw allow ssh
```

Comment:
SSH must be enabled in all servers. I'm not sure how you are enabling SSH so perhaps this command is redundant.